ui/shared/components/form/Inputs.jsx

```diff
@@ -18,23 +19,7 @@ export class BaseSemanticInput extends React.Component {
   }
 
   shouldComponentUpdate(nextProps, nextState) {
-    const { options } = this.props
-    if (nextProps.options) {
-      if (nextProps.options.length !== (options || []).length) {
-        return true
-      }
-      Object.entries(nextProps.options).forEach(([i, opt]) => { // eslint-disable-line consistent-return
-        if (['value', 'text', 'color', 'disabled', 'description'].some(k => opt[k] !== options[i][k])) {
-          return true
-        }
-      })
-    }
-    if (Object.keys(nextProps).filter(k => k !== 'onChange' && k !== 'options').some(
-      k => nextProps[k] !== this.props[k], // eslint-disable-line react/destructuring-assignment
-    )) {
-      return true
-    }
-    return nextState !== this.state
+    return semanticShouldUpdate(this, nextProps, nextState)
```

Comment:
the name of this function is confusing, maybe name it something like `formInputComponentShouldUpdate`